### ISSUE TITLE
LG-37: Jaeger support in MicroProfile Open Tracing

### ISSUE THREAD

**Issue by fmhwong** [OpenLiberty/open-liberty#6906]
AHA: https://bigblue.aha.io/features/LG-37

~~ID: https://github.ibm.com/was-liberty/liberty-docs/issues/1143~~
ID: https://github.com/OpenLiberty/docs/issues/310

**Comment by marikaj123** [comment]
@fmhwong - Did the feature make it into Liberty beta 190010 release?

**Comment by marikaj123** [comment]
If the feature is planned to GM for the next release, please add the label target:ga.

**Comment by marikaj123** [comment]
Going through WAD Review next week.

**Comment by Emily-Jiang** [comment]
WAD review comments:
1. p5: opentracing io -> openliberty.io
2. p8: I think Thorntail supports both Jaeger and Zipkin. Felix to double check
3. p15: Emily asks whether we continue maintaining zipkin user feature. Felix said if MicroProfile adopts opentracing 0.34 then we will update zipkin user feature
4. Talk to YK regarding creating a new guide for Jeager support
5. have a discussion to see whether there is a risk to embed this feature in an existing mpOpenTracing-1.3

**Comment by marikaj123** [comment]
@fmhwong - are we planning to release this feature as a beta in 190011?

**Comment by fmhwong** [comment]
> WAD review comments:
> 
>     1. p5: opentracing io -> openliberty.io
Done
> 
>     2. p8: I think Thorntail supports both Jaeger and Zipkin. Felix to double check
Thorntail embedded Jaeger client in their runtime.
> 
>     3. p15: Emily asks whether we continue maintaining zipkin user feature. Felix said if MicroProfile adopts opentracing 0.34 then we will update zipkin user feature
The version should be 0.33.
> 
>     4. Talk to YK regarding creating a new guide for Jeager support
https://github.com/OpenLiberty/guides-common/issues/360
> 
>     5. have a discussion to see whether there is a risk to embed this feature in an existing mpOpenTracing-1.3
> 
>     6. P27 add encrypted and decoded password, while preferring encrypted approach
Updated documentation.
> 
>     7. P29: take to PoC on the request of changing message ID from E to I when there is no tracer implementation while having mpOpenTracing feature enabled
Will be discussed on 12/23/2019
> 
>     8. Testing -add SVT to the Test page
Done.

**Comment by fmhwong** [comment]
- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
 ## List of Steps to complete or get approvals / sign-offs for Onboarding to the Liberty release (GM date)
 
Instructions:
 - Do the actions below and mark them complete in the checklist when they are done.
 - Make sure all feature readiness approvers put the appropriate tag on the epic to indicate their approval.
 
- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
 ## TARGET COMPLETION DATE **Before Development Starts or 8 weeks before Onboarding** 
- [x] POC Design / WAD Review Scheduled (David Chang) or N/A. AHA Feature marked.
- [x] POC Design / WAD Reviewed (Feature Owner) or N/A. AHA Feature marked.
- [x] Complete any To-Do's from the POC Review, and create a new TO-DO in the AHA Feature for the "Design Approval Request" and assign it to Ian Robinson. Leave the date blank. (Feature Owner)
- [ ] Design / WAD Approval (Ian Robinson) or N/A. AHA Feature marked.
- [ ] No Design / No WAD Approval (Arthur De Magalhaes - cloud / Alasdair Nottingham - server) or N/A. AHA Feature marked.
- [x] AHA Feature has link to Github epic or issue. Github epic has link back to AHA Feature. (Feature Owner or Backlog Subtribe PM).
- [x] SVT Requirements identified. (Epic owner / Feature owner with SVT focal point)
- [x] ID Requirements identified. (Epic owner / Feature owner with ID focal point)
- [x] Create a child task of the epic entitled "FAT Approval Test Summary"
 
- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
 ## TARGET COMPLETION DATE **3 weeks before Onboarding** 
- [x] Identify all open source libraries that are changing or are new. Work with Legal Release Services (Cass Tucker or Release PM) to get open source cleared and approved. Or N/A. (Epic Owner).   New or changed open source impacts license and Certificate of Originality.
 
- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
 ## TARGET COMPLETION DATE ** 3 weeks before Onboarding** 
- [x] All new or changed PII messages are checked into the integration branch, before the last translation shipment out. (Epic Owner)
 
- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
 ## TARGET COMPLETION DATE **2 weeks before Onboarding** 
- [x] Implementation complete. (Epic owner / Feature owner) 
- [x] All function tests complete. Ready for FAT Approval. (Epic owner / Feature owner)
- [ ] Review all known issues for Stop Ship. (Epic owner / Feature owner / PM)
 
- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
 ## APPROVALS with TARGET COMPLETION DATE **2 to 1 week before Onboarding** 
Prereq: You must have the Design Approval or No Design Approval on the AHA Feature.
- [ ] **Accessibility** - (G Scott Johnston). Accessibility testing is complete or N/A. Approver adds label focalApproved:accessibility to the Epic in Github.
- [ ] **FAT Liberty SOE** - (Kevin Smith). SOE FATS are running successfully or N/A . Approver adds label focalApproved:fat to the Epic in Github.
- [ ] **Globalization** (Marika Joannidis - Liberty / Simy Cheeran - tWAS). Translation is complete or N/A. TVT - complete or N/A. Approver adds label focalApproved:globalization to the Epic in Github.
- [ ] **ID** - (Kareen Deen). Documentation work is complete or N/A . Approver adds label focalApproved:id to the Epic in Github.
- [ ] **Performance** - (Jared Anderson). Performance testing is complete with no high severity defects or N/A . Approver adds label focalApproved:performance to the Epic in Github.
- [ ] **Serviceability** - (Don Bourne). Serviceability has been addressed. 
- [ ] **STE** - (Swati Kasundra). STE chart deck is complete or N/A . Approver adds label focalApproved:ste to the Epic in Github.
- [ ] **SVT** - (Tanya Kulik). SVT is complete or N/A . Approver adds label focalApproved:svt to the Epic in Github.
 
- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
## TARGET COMPLETION DATE **1 week before Onboarding** 
- [ ] No Stop Ship issues for the feature. (Epic owner / Feature owner / Release PM)
- [ ] Ship Readiness Review and Release Notes completed (Epic owner / Feature owner / Release PM)
- [ ] Github Epic and Epic's issues are closed / complete. All PRs are committed to the master branch. (Epic owner / Feature owner / Backlog Subtribe PM)
 
- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
 ## NOT REQUIRED FOR A FEATURE
- [x] **OL Guides** - (Yee-Kang Chang). Assessment for OL Guides is complete or N/A. 
- [ ] **WDT** - (Leonard Theivendra). WDT work complete or N/A.
 
 
- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
## Related Deliverables TARGET COMPLETION DATE **General Availability** 
- [ ] Blog article writeup (Epic owner / Feature owner / Laura Cowen)

**Comment by fmhwong** [comment]
@gscottj This feature doesn't have any UI.  May I have your approval on accessibility?

**Comment by fmhwong** [comment]
@jhanders34 May I have your approval on performance?

**Comment by fmhwong** [comment]
@marikaj123 May I have the globalization approval?

**Comment by fmhwong** [comment]
@chirp1 ID and blog have been prepared.  May I have the ID approval?

**Comment by marikaj123** [comment]
According to Felix tag, the beta code did make it into the Beta release 190011

**Comment by marikaj123** [comment]
Approving since the translated files are in Master

**Comment by fmhwong** [comment]
Design document: https://ibm.box.com/s/uekfr2k7aeqe8r6xuvpd15v78vt6ge85

**Comment by fmhwong** [comment]
@skasund STE for this feature has been uploaded to box folder.  Please approve for STE.  Thanks!

**Comment by skasund** [comment]
@fmhwong  Thanks for the STE slides.  I've provided my approval.

**Comment by fmhwong** [comment]
@sabolo Hi Kevin, this epic is ready for FAT approval.

**Comment by tkulik** [comment]
From SVT perspective Liberty serviceability for this feature looks good.

**Comment by fmhwong** [comment]
@sabolo @dave-waddling Please review for FAT approval.  Thanks!

**Comment by fmhwong** [comment]
@donbourne
**Serviceability Approval Comment** - Please answer the following questions for serviceability approval:

1.  WAD -- does the WAD identify the most likely problems customers will see and identify how the feature will enable them to diagnose and solve those problems without resorting to raising a PMR?  Have these issues been addressed in the implementation?

- Yes, most likely the Jaeger client jars are missing from the WAR lib, not defined as shared library, or defined in shared library but the jars files are not in that location.  In these cases, an error message will be printed to indicate tracing client libraries is not in classpath.

2.  Test and Demo -- As part of the serviceability process we're asking feature teams to test and analyze common problem paths for serviceability and demo those problem paths to someone not involved in the development of the feature (eg. L2, test team, or another development team).  
a) What problem paths were tested and demonstrated?
b) Who did you demo to?
c) Do the people you demo'd to agree that the serviceability of the demonstrated problem scenarios is sufficient to avoid PMRs for any problems customers are likely to encounter, or that L2 should be able to quickly address those problems without need to engage L3?

- During SVT, there were cases where the Jaeger client library was not properly defined, and the error message helped in that situation.

3.  SVT -- SVT team is often the first team to try new features and often encounters problems setting up and using them.  Note that we're not expecting SVT to do full serviceability testing -- just to sign-off on the serviceability of the problem paths they encountered.
a) Who conducted SVT tests for this feature?

- This was tested by SVT Tanya Kulik 


b) Do they agree that the serviceability of the problems they encountered is sufficient to avoid PMRs, or that L2 should be able to quickly address those problems without need to engage L3?  

-  Tanya has no concern on serviceability.

4.  Which L2 / L3 queues will handle PMRs for this feature?  Ensure they are present in the contact reference file and in the queue contact summary, and that the respective L2/L3 teams know they are supporting it.  Ask Don Bourne if you need links or more info.

- WAS L3: Liberty Log Analytics

**Comment by chirp1** [comment]
OL and WL doc issues are open. Approving.

**Comment by marikaj123** [comment]
Felix, Thank you for confirming the feature did make the GM release 190012. When you can please close the feature.  Thank you.

**Comment by fmhwong** [comment]
Shipped in 19.0.0.12
